Neos.Media/Classes/Domain/Service/ThumbnailService.php

```diff
@@ -92,11 +92,6 @@ public function getThumbnail(AssetInterface $asset, ThumbnailConfiguration $conf
             if ($asset->getWidth() === null && $asset->getHeight() === null) {
                 return $asset;
             }
-            $maximumWidth = ($configuration->getMaximumWidth() > $asset->getWidth()) ? $asset->getWidth() : $configuration->getMaximumWidth();
-            $maximumHeight = ($configuration->getMaximumHeight() > $asset->getHeight()) ? $asset->getHeight() : $configuration->getMaximumHeight();
-            if ($configuration->isUpScalingAllowed() === false && $maximumWidth === $asset->getWidth() && $maximumHeight === $asset->getHeight()) {
-                return $asset;
-            }
```

Comment:
there's another place that does the same? where?

Well that's definitely a matter of perspective, I'm sure when this was first implemented it was meant as a feature to prevent generating identical thumbnails. I personally would prefer it wasn't here, but would some more opinions on this.

don't see why the code would become bloated, aside that this method already is quite bloated itself.. but could be moved to a separate method and the simply have a configuration injected boolean property checking..